WordPress/gutenberg
# [RNMobile] Consolidate inserter block filter

## Description

This consolidates how available blocks are filtered in the inserter menu. Currently the block list is filtered the same way in two different components. Instead this adds a single filter function to the first common ancestor where it is then passed down to the each child.

The intent of the common filter is to prevent drift between the various inserter menu block lists, allow for easy extension when later blocks require a change in availability, and for scoping reusable blocks.

Fixes #34303

## How has this been tested?



1. Open the editor where reusable blocks are available.
2. Tap the inserter in the bottom toolbar
3. Verify that the default block list only shows what blocks are available*
4. Switch to the reusable tab
5. Verify that only reusable blocks are available
6. Activate the inserter search
7. Verify that the initial search results  includes reusable blocks at the bottom of the list (i.e. search activated but with no search query)
8. Verify that reusable blocks can be found by searching for them by name


## Screenshots 

## Types of changes





## Checklist:
